docs/source/torch.compiler_faq.rst

```diff
@@ -614,23 +614,58 @@ to tweak our ``numpy_fn`` so that it accepts cuda Tensors and returns tensors:
 .. code-block:: python
 
    @torch.compile
-   def numpy_fn(X: torch.Tensor, Y: torch.Tensor) -> torch.Tensor:
+   def torch_fn(X: torch.Tensor, Y: torch.Tensor) -> torch.Tensor:
        X, Y = X.numpy(), Y.numpy()
-       Z = np.sum(X[:, :, None] * Y[:, None, :], axis=(-2, -1))
+       Z = numpy_fn(X, Y)
        return torch.from_numpy(Z)
 
    X = torch.randn(1024, 64, device="cuda")
    Y = torch.randn(1024, 64, device="cuda")
    with torch.device("cuda"):
-       Z = numpy_fn(X, Y)
+       Z = torch_fn(X, Y)
 
 By doing this, we explicitly create the tensors in CUDA memory, and we keep
 them there. In this case ``X.numpy()`` and ``from_numpy()`` are hints to the compiler
 but no real data movement happens. Note that the original program would not run
 on eager mode now. If you want to run it in eager mode, you would need to call
 ``.numpy(force=True)`` doing ``Z = Z.cuda()`` before returning
 ``Z``. Of course, doing this would execute the program on eager mode NumPy, and
-on CPU.
+on CPU. Note also that calling ``.numpy(force=True`` under ``torch.compile`` also
+moves to CPU and deactivates gradient computations.
+
+We provide the decorator ``torch.compiler.wrap_numpy``  that implements the patteron of wrapping
```

Comment:
I would move this at the top actually. Users should use this instead of doing it by hand in most cases?